File: frontend/src/services/dashboardService.js
import api from './api'

export default {
  getStats() {
    return api.get('/admin/dashboard/stats')
  },
  getActivities(page = 0, size = 10, type = 'ALL') {
    return api.get('/admin/dashboard/activities', { params: { page, size, type } })
  },
}

File: backend/src/main/java/sn/immosn/backend/dashboard/service/DashboardService.java
package sn.immosn.backend.dashboard.service;

import sn.immosn.backend.client.web.dashboard.dto.DashboardStatsDto;
import sn.immosn.backend.client.web.dashboard.dto.RecentActivityDto;
import sn.immosn.backend.shared.response.PagedResponse;

public interface DashboardService {
    DashboardStatsDto getStats(String callerEmail);
    PagedResponse<RecentActivityDto> getActivities(int page, int size, String type, String callerEmail);
}
